### ISSUE TITLE
Some adoc files having reference issue

### ISSUE THREAD

**Issue by TangoEnSkai** [circleci/circleci-docs#7226]
# Description
- some docs in `*.adoc` seems having URL reference issue

# Location

for example:
- docs: https://circleci.com/docs/using-dynamic-configuration
- file: https://github.com/circleci/circleci-docs/blob/master/jekyll/_cci2/using-dynamic-configuration.adoc

you may see the rendered doc looks like this:

```
Define a job called setup that uses the continuation orb as an [executor]({{ site.baseurl }}/executor-intro/). 
```

or

![Screenshot 2022-08-08 at 18 26 28](https://user-images.githubusercontent.com/21152231/183386355-e4dc949d-91a9-4131-8565-936001f87a90.png)

The URL of executor doc, which supposed to be `site.baseurl/docs/executor-intro` or `https://circleci.com/docs/executor-intro`
is unintentionally rendered.

# Steps to Reproduce

1. go to https://circleci.com/docs/using-dynamic-configuration
2. search `/docs/executor-intro/` (e.g. cmd + f then put `/docs/executor-intro/`)

**Comment by TangoEnSkai** [comment]
This might be out of topic but I am curious if there any background why we use both 
- `md`
- `adoc`

under `https://github.com/circleci/circleci-docs/tree/master/jekyll/_cci2`

because I reckon, using several formats may causes issue to maintain docs due to these style nits.

**Comment by devinmarieb** [comment]
@TangoEnSkai We are migrating away from markdown as asciidoc offers a lot more flexibility and has more extensive customizations available out of the box. Originally we used asciidoc for Server because it was much easier to convert those docs to PDF that way. Now we are adopting it everywhere, but we can't do a batch conversion because of the translation workflow issues it would create.

**Comment by TangoEnSkai** [comment]
> @TangoEnSkai We are migrating away from markdown as asciidoc offers a lot more flexibility and has more extensive customizations available out of the box. Originally we used asciidoc for Server because it was much easier to convert those docs to PDF that way. Now we are adopting it everywhere, but we can't do a batch conversion because of the translation workflow issues it would create.


Hi, @devinmarieb 
understood and thanks for the fix #7228 👍 

![Screenshot 2022-08-09 at 11 29 53](https://user-images.githubusercontent.com/21152231/183550519-b9e8d80a-3cb8-4fcf-abd1-82e120eea8d6.png)

now the doc seems good with your fix, I will do the same if I found another.
